pkg/query-service/app/integrations/thirdPartyApi/translator.go

```diff
@@ -1,598 +1,514 @@
 package thirdPartyApi
 
 import (
+	"fmt"
 	"net"
+	"time"
 
-	v3 "github.com/SigNoz/signoz/pkg/query-service/model/v3"
+	qbtypes "github.com/SigNoz/signoz/pkg/types/querybuildertypes/querybuildertypesv5"
+	"github.com/SigNoz/signoz/pkg/types/telemetrytypes"
 )
 
 const (
-	urlPathKey    = "http.url"
-	serverNameKey = "net.peer.name"
+	urlPathKeyLegacy    = "http.url"
+	serverNameKeyLegacy = "net.peer.name"
+
+	urlPathKeyCurrent    = "url.full"
+	serverNameKeyCurrent = "server.address"
 )
 
-var defaultStepInterval int64 = 60
+var defaultStepInterval = 60 * time.Second
+
+// SemconvFieldMapping defines the mapping between legacy and current semconv
+type SemconvFieldMapping struct {
+	LegacyField  string
+	CurrentField string
+	FieldType    telemetrytypes.FieldDataType
+	Context      telemetrytypes.FieldContext
+}
+
+var semconvMappings = []SemconvFieldMapping{
+	{
+		LegacyField:  urlPathKeyLegacy,
+		CurrentField: urlPathKeyCurrent,
+		FieldType:    telemetrytypes.FieldDataTypeString,
+		Context:      telemetrytypes.FieldContextAttribute,
+	},
+	{
+		LegacyField:  serverNameKeyLegacy,
+		CurrentField: serverNameKeyCurrent,
+		FieldType:    telemetrytypes.FieldDataTypeString,
+		Context:      telemetrytypes.FieldContextAttribute,
+	},
+}
+
+// CreateDualSemconvGroupByKeys creates group by keys supporting both semconv versions
+func CreateDualSemconvGroupByKeysServer() []qbtypes.GroupByKey {
+	return createDualSemconvGroupBy(semconvMappings[1]) // server name mapping
+}
+
+func CreateDualSemconvGroupByKeysUrl() []qbtypes.GroupByKey {
+	return createDualSemconvGroupBy(semconvMappings[0]) // url path mapping
```

Comment:
I feel this part is unnecessary added complexity to the code, please make it simpler.

You can create a map of the server and url , and call them using name "server" and "url" during group by. It will be simpler.